vuu-ui/packages/vuu-shell/src/application-settings/ApplicationSettingsPanel.tsx

```diff
@@ -1,58 +1,193 @@
 import { getFieldName, queryClosest } from "@finos/vuu-utils";
-import { useComponentCssInjection } from "@salt-ds/styles";
-import { useWindow } from "@salt-ds/window";
-import { FormEventHandler, HTMLAttributes, useCallback } from "react";
-
-import applicationSettingsPanelCss from "./ApplicationSettingsPanel.css";
 import {
+  Dropdown,
   FormField,
   FormFieldLabel,
+  Input,
+  Option,
+  Switch,
   ToggleButton,
   ToggleButtonGroup,
 } from "@salt-ds/core";
-import { useApplicationSettings } from "../application-provider";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import {
+  SingleSelectionHandler,
+} from "@finos/vuu-ui-controls";
+import { FormEventHandler, HTMLAttributes, useCallback } from "react";
+
+import applicationSettingsPanelCss from "./ApplicationSettingsPanel.css";
+
+// Type definitions
+export type SettingsProperty<
+  T extends string | number | boolean | object = string
+> = {
+  name: string;
+  label: string;
+  values?: T[] | object;
+  defaultValue?: T;
+  type: "string" | "boolean" | "number";
+};
+
+export interface SettingsSchema {
+  properties: SettingsProperty[];
+}
+
+// Determine the form control type to be displayed
+export function getFormControl(
+  property: SettingsProperty,
+  changeHandler: FormEventHandler,
+  selectHandler: SingleSelectionHandler,
```

Comment:
this is not the right type for the salt dropdown

the selection handler looks like this

```
(evt: SyntheticEvent, newSelected: string[]) => void
```

If you hover over `onSelectinChange` in the line below, intellisense will show you the correct type
```

You can avoid coding the type longhand, if you wish, by directly referencing the original type in the DropdownProps, like this

```
selectHandler: DropdownProps['onSelectionChange']
```


<Dropdown value={currentValue} onSelectionChange={selectHandler}>
```